Running test suite locally does not work on M1 Mac

### Problem

After running `bin/fetch-golangci-lint` and `bin/run-tests` on an M1 Mac (or other arm64 system), contributors receive the following error:

```
copy exemplar exercises/practice/word-search/.meta/example.go to tmp/lint/practice/word_search
copy module exercises/practice/word-search/go.mod to tmp/lint/practice/word_search
copy generator exercises/practice/word-search/.meta/gen.go
put everything in the same module
copy base files and tests to tmp/lint/practice/wordy
copy exemplar exercises/practice/wordy/.meta/example.go to tmp/lint/practice/wordy
copy module exercises/practice/wordy/go.mod to tmp/lint/practice/wordy
copy generator exercises/practice/wordy/.meta/gen.go
put everything in the same module
copy base files and tests to tmp/lint/practice/yacht
copy exemplar exercises/practice/yacht/.meta/example.go to tmp/lint/practice/yacht
copy module exercises/practice/yacht/go.mod to tmp/lint/practice/yacht
copy generator exercises/practice/yacht/.meta/gen.go
put everything in the same module
copy base files and tests to tmp/lint/practice/zebra_puzzle
copy exemplar exercises/practice/zebra-puzzle/.meta/example.go to tmp/lint/practice/zebra_puzzle
copy module exercises/practice/zebra-puzzle/go.mod to tmp/lint/practice/zebra_puzzle
put everything in the same module
running linter and tests for exercises and exercise generators
panic: load embedded ruleguard rules: rules/rules.go:13: can't load fmt

goroutine 1 [running]:
github.com/go-critic/go-critic/checkers.init.22()
    github.com/go-critic/go-critic@v0.6.3/checkers/embedded_rules.go:47 +0x494
LINTER/TESTS FAILED for exercises or exercise generators
deleting tmp/lint
```

This is due to an interaction with `golangci-lint` and `go-critic` on older versions of `golangci-lint`. 
See [this issue](https://github.com/golangci/golangci-lint/issues/3107) for more information.

Note, this issue has been [documented](https://github.com/golangci/golangci-lint/issues/2673) for arm64 systems with certain versions of Go, but doesn't seem to occur on amd64 architectures. However, it doesn't appear to be an issue for arm64 anymore since v1.49.0.

### Steps to reproduce

1. On an M1 mac or other arm64 machine, run the current `bin/fetch-golangci-lint` script
2. Try running the tests with `bin/run-tests`
3. See error above

--- a/README.md
+++ b/README.md
@@ -22,7 +22,7 @@ You will need a github account and you will need to fork exercism/go to your acc
 See [GitHub Help](https://help.github.com/articles/fork-a-repo/) if you are unfamiliar with the process.
 Clone your fork with the command: `git clone https://github.com/<you>/go`.
 Test your clone by cding to the go directory and typing `bin/fetch-golangci-lint` and then
-`bin/test-without-stubs`. You should see tests pass for all exercises.
+`bin/run-tests`. You should see tests pass for all exercises.
 
 Note that unlike most other Go code, it is not necessary to clone this to your GOPATH.
 This is because this repo only imports from the standard library and isn't expected to be imported by other packages.

--- a/bin/fetch-golangci-lint
+++ b/bin/fetch-golangci-lint
@@ -1,3 +1,3 @@
 #!/bin/bash
 
-curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.46.2
+curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.49.0

